boards/opentitan/src/tests/otbn.rs

```diff
@@ -66,13 +67,13 @@ impl<'a> OtbnTestCallback {
 impl<'a> Client<'a> for OtbnTestCallback {
     fn op_done(&'a self, result: Result<(), ErrorCode>, output: &'static mut [u8]) {
         self.op_done.set(true);
+        self.output_buf.replace(output);
         assert_eq!(result, Ok(()));
-        assert_eq!(output[0..256], EXPECTING);
+        // Use this copy of output for assertion
+        assert_eq!(self.output_buf.take().unwrap()[0..256], EXPECTING);
```

Comment:
Close, but this won't work.

You correctly replace the `self.output_buf` above, but then you take it here and it gets dropped.

You would instead want to do:

```rust
assert_eq!(output[0..256], EXPECTING);
self.output_buf.replace(output);
```